scripts/core/menu/menu.js

```diff
@@ -31,7 +31,14 @@ angular.module('superdesk.core.menu', [
             this.flags = superdeskFlags.flags;
 
             this.openUpload = function openUpload(files) {
```

Comment:
This function can take extra parameters (you may look into the [official docs](https://github.com/danialfarid/ng-file-upload#file-select-and-drop)). You can capture the drop event here, but you will have to make a change in `scripts/core/menu/views/superdesk-view.html`:
```html
<div id="main-container" ng-class="superdesk.flags" ng-if="!popup && superdesk.flags.workspace === true" ngf-drop="superdesk.openUpload($files, $event)">
```
Notice the second `$event` param I've added at the very end of the line. Now, you can change this line to:
```js
this.openUpload = function openUpload(files, event) {
```
and use [`event.target.contains`](https://developer.mozilla.org/en-US/docs/Web/API/Node/contains) function to find out if the drop target is `div.public-DraftEditor-content`, in which case do not upload. It would also be nice to leave a comment like `// Editor3 handles its own file drop events`.

Then you can use the editor3 actions to handle the upload, without using these global variables. Just like it works now.